registration-processor/pre-processor/registration-processor-osi-validator-stage/src/main/java/io/mosip/registration/processor/stages/osivalidator/OSIValidator.java

```diff
@@ -1,781 +1,857 @@
+	private static final String ISTRUE = "true";
+
+	@Value("${mosip.kernel.applicant.type.age.limit}")
+	private String ageLimit;
+
+	@Value("${registration.processor.applicant.dob.format}")
+	private String dobFormat;
+
+	@Value("${registration.processor.validate.introducer}")
+	private boolean introducerValidation;
+
+	@Value("${registration.processor.sub-process}")
+	private String subProcesses;
+	
+	@Value("${registration.processor.validateIndividualBiometricsforSubProcess.enabled}")
+	private boolean validateIndividualBiometricsforSubProcess;
+	
+	@Value("${registration-processor.validateOfficerOrSupervisorBiometricsforSubProcess.enabled}")
+	private boolean validateOfficerOrSupervisorBiometricsforSubProcess;
+	
+	
+	
+	@Autowired
+	private Utilities utility;
+
+	@Autowired
+	RegistrationExceptionMapperUtil registrationExceptionMapperUtil;
+
+	@Autowired
+	private AuthUtil authUtil;
+
+	/** The Constant APPLICATION_ID. */
+	public static final String IDA_APP_ID = "IDA";
+
+	/** The Constant RSA. */
+	public static final String RSA = "RSA";
+
+	/** The Constant RSA. */
+	public static final String PARTNER_ID = "PARTNER";
+
+	public static final String INDIVIDUAL_TYPE_UIN = "UIN";
+
+	private static final String INDIVIDUAL_TYPE_USERID = "USERID";
+	
+	private static final String APPID = "regproc";
+
+	@Autowired
+	ObjectMapper mapper;
+	/**
+	 * Checks if is valid OSI.
+	 *
+	 * @param registrationId the registration id
+	 * @return true, if is valid OSI
+	 * @throws IOException                           Signals that an I/O exception
+	 *                                               has occurred.
+	 * @throws ApisResourceAccessException           the apis resource access
+	 *                                               exception
+	 * @throws SAXException
+	 * @throws ParserConfigurationException
+	 * @throws NoSuchAlgorithmException
+	 * @throws InvalidKeySpecException
+	 * @throws BioTypeException
+	 * @throws BiometricException
+	 * @throws NumberFormatException
+	 * @throws                                       io.mosip.kernel.core.exception.IOException
+	 * @throws PacketDecryptionFailureException
+	 * @throws ParentOnHoldException
+	 * @throws AuthSystemException
+	 * @throws RegistrationProcessorCheckedException
+	 */
+	public boolean isValidOSI(String registrationId, InternalRegistrationStatusDto registrationStatusDto, Map<String, String> metaInfo)
+			throws IOException, ApisResourceAccessException, InvalidKeySpecException, NoSuchAlgorithmException,
+			NumberFormatException, BiometricException, BioTypeException,
+			AuthSystemException, PacketManagerException, JSONException, JsonProcessingException, ParentOnHoldException, CertificateException {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+				registrationId, "OSIValidator::isValidOSI()::entry");
+		boolean isValidOsi = false;
+		
+		
+		if(subProcesses.contains(registrationStatusDto.getRegistrationType())) {

```

Comment:
dont use .contains
because there could be process and subprocess which might be matching. SUppose we have process  NEWREGISTRATION
and subprocess NEW
in this case it will fail. Actually now we dont know how many process and subprocess will e there. So its always better to do exact match, U can write a method in one util in core which can be used everywhere.